### PR TITLE
Move some `api` package functions away

### PR DESCRIPTION
**Pull request by vdemeester**
- DisplayablePorts is a `cli` function, moving to `docker/cli`
- Move MatchesContentType to the only package using it,
  `api/server/httputils` (and remove the deps on logrus for `api` package)

Related PR in docker/cli : https://github.com/docker/cli/pull/236

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: api/server/httputils/httputils.go [hunk 1]
```diff
@@ -8,7 +8,9 @@ import (
 
 	"golang.org/x/net/context"
 
-	"github.com/docker/docker/api"
+	"mime"
```
Comment (thaJeztah) [inline api/server/httputils/httputils.go]: should this go to the top?

**Review by thaJeztah — APPROVED** [review]
one nit, but LGTM

File: api/server/httputils/httputils.go (hunk 1 above)
Comment (vdemeester) [inline api/server/httputils/httputils.go]: ahah yes it should 😛

**Review by dnephin — APPROVED** [review]
LGTM with moved import

File: api/server/httputils/httputils.go [hunk 2]
```diff
@@ -86,3 +88,12 @@ func VersionFromContext(ctx context.Context) string {
 
 	return ""
 }
+
+// MatchesContentType validates the content type against the expected one
+func MatchesContentType(contentType, expectedType string) bool {
```
Comment (cpuguy83) [inline api/server/httputils/httputils.go]: Does this need to be exported?

Comment (vdemeester) [inline api/server/httputils/httputils.go]: Probably not 😇

**Comment by vdemeester** [comment]
@cpuguy83 @dnephin updated 👼

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Review by mlaventure — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
docker-py looks to fail on experimental and powerpc;

```
01:41:31 =================================== FAILURES ===================================
01:41:31 _________________ ServiceTest.test_create_service_with_secret __________________
01:41:31 /docker-py/tests/integration/api_service_test.py:429: in test_create_service_with_secret
01:41:31     container = self.get_service_container(name)
01:41:31 /docker-py/tests/integration/api_service_test.py:41: in get_service_container
01:41:31     all=include_stopped
01:41:31 /docker-py/docker/api/container.py:189: in containers
01:41:31     res = self._result(self._get(u, params=params), True)
01:41:31 /docker-py/docker/utils/decorators.py:46: in inner
01:41:31     return f(self, *args, **kwargs)
01:41:31 /docker-py/docker/api/client.py:183: in _get
01:41:31     return self.get(url, **self._set_request_timeout(kwargs))
01:41:31 /usr/lib/python2.7/dist-packages/requests/sessions.py:469: in get
01:41:31     return self.request('GET', url, **kwargs)
01:41:31 /usr/lib/python2.7/dist-packages/requests/sessions.py:457: in request
01:41:31     resp = self.send(prep, **send_kwargs)
01:41:31 /usr/lib/python2.7/dist-packages/requests/sessions.py:569: in send
01:41:31     r = adapter.send(request, **kwargs)
01:41:31 /usr/lib/python2.7/dist-packages/requests/adapters.py:407: in send
01:41:31     raise ConnectionError(err, request=request)
01:41:31 E   ConnectionError: ('Connection aborted.', BadStatusLine("''",))
01:41:31 ========= 1 failed, 244 passed, 8 skipped, 3 xfailed in 394.97 seconds =========
01:41:31 ---> Making bundle: .integration-daemon-stop (in bundles/17.06.0-dev/test-docker-py)
01:41:31 +++++ cat bundles/17.06.0-dev/test-docker-py/docker.pid
01:41:31 ++++ kill 14936
```

not sure if it's related, but both fail on the same test, so could be

**Comment by vdemeester** [comment]
@thaJeztah Restarted the build and it's now green.. Seems unrelated 😓

**Pull request merged**